tabby-ssh/src/components/sshTab.component.ts

```diff
@@ -157,24 +157,26 @@ export class SSHTabComponent extends BaseTerminalTabComponent<SSHProfile> implem
     protected attachSessionHandlers (): void {
         const session = this.session!
         this.attachSessionHandler(session.destroyed$, () => {
-            if (
-                // Ctrl-D
-                this.recentInputs.charCodeAt(this.recentInputs.length - 1) === 4 ||
```

Comment:
Did this functionality get lost here? For SSH, there's an important distinction: you never want to keep the session open after you explicitly kill it with Ctrl-D or `exit`

Reply:
Oh yeah i can see your point! In that case, wouldn't that be more adequate to create two distinct options :

- Close the tab on session end => only when explicitly terminated (default and actual behaviour) / always / no

- For Serial, Telnet and SSH, automatically reconnect => yes/no *(locked if tabs are always closed on session end)*
(do not try to reconnect if the session has been manually closed even if yes selected)

It would probably be a good comprise for those who want to keep a way to scroll in the log after the session has been terminated (manually or not) and those who want the tabs to close when the connection is lost.

Reply:
What if we do it the same way iTerm does it with three options (close / keep / reconnect) + a new one ("auto") where `exit` / Ctrl-D is "close" and otherwise "keep"?

And if you implement this on a profile level, then it's possible to set sensible defaults for each session type (close for local, auto for others)

Reply:
Ok let's do this !
I'm converting this PR into draft until it is ready to review again.